src/status_im/contexts/wallet/common/utils.cljs

```diff
@@ -269,3 +270,10 @@
     (some #(when (= token-symbol (:symbol %))
              %)
           address-tokens)))
+
+(defn get-account-name-error
```

Comment:
I tend to think that we can reduce amount of code duplication by creating a more generic error function for accounts that has logic related to names, colors, emoji etc. and produce single error code. Something like that:
```clojure
(defn account-uniqueness-error
  [name color emoji]
  (let [accounts-names              (rf/sub [:wallet/accounts-names])
        accounts-emojis-and-colors  (rf/sub [:wallet/accounts-emojis-and-colors])
        ...]
    (cond
      (utils.string/contains-emoji? s)             :emoji
      (accounts-names s)                   :existing-name
      (utils.string/contains-special-character? s) :special-character
      (repeated? (accounts-emojis-and-colors [emoji color])) :duplicated-emoji-color)))
```
In the views we can have one-liner that depends on all props and constantly rechecks errors
```clojure
(let [error (utils/account-uniqueness-error name color emoji)])
```

This way we won't need explicit calls on every emoji or color or name change like `(check-emoji-and-color-error @emoji new-color)`, because if something is wrong, error already contains information about it.

Wdyt?

Reply:
Hey @vkjr 

Thanks for the suggestion.

A problem is that we still need to handle in a different way the errors for the user for each screen since they have significant differences (e.g. in edit screen the emoji+color change is performed "on change", no need to hit the confirm button).

When I tried to unify the errors in a single atom I faced bugs, or validations being cleared, or had to exhaustively  recheck for the errors with less readable code. So I decided to keep two error atoms, one for the input, and other for the emoji+color.

I think the current implementation is more verbose but it's more robust and easier to use in the handlers triggered when something is changed. So this time I'd prefer to keep the current approach, maybe in a follow up we could explore ways to abstract the behavior?